data_processing/data_treatment.py

```diff
@@ -0,0 +1,218 @@
+    return df
+
+def fill_with_zero(df, pattern, regexp = False):
+    """Fills columns given str patter or regex
+    
+    Arguments:
+        df {DataFrame} -- Pandas DataFrame
+        pattern {str} -- String pattern
+    
+    Keyword Arguments:
+        regexp {bool} -- Pattern is about a string or not (default: {False})
+    
+    Returns:
+        DataFrame -- DataFrame with given/selected columns filled with 0
+    """
+    col_list_selector = df.columns.str.contains( pattern , regex = regexp)
+    df.loc[:,col_list_selector] = df.loc[:,col_list_selector].fillna(value = 0)
+
+    return df
+
+
+
+
+def treat_datetime(df, col):
+    """Treats date or datetime columns to machine learning ready vals
+    
+    Arguments:
+        df {DataFrame} -- Pandas DataFrame
+        col {str} -- Datetime or timestamp column name
+    
+    Returns:
+        DataFrame -- Full DataFrame with year column and date column converted to numerical
+    """
+    df[col+"_numeric"] = pd.to_numeric(pd.to_datetime(df[col]))/100000000000
+    df[col+"_year"] = pd.to_datetime(df[col]).dt.year
+    
+    return df
+
+def map_dataframe(df, mapping_dict):
+    for k, v in mapping_dict.items():
+        df[k] = df[k].map(v)
+    return df
+
+def fill_by_dict(df, filling_dict):
+    for k, v in filling_dict.items():
+        method = v["method"]
+        if method == "constant":
+            df[k].fillna(method = None, value = v["value"], inplace = True)
+        elif method == "mode":
+            df[k].fillna(method = None, value = df[k].mode()[0], inplace = True)
+        elif method == "mean":
+            df[k].fillna(df[k].mean(), inplace = True)
+        elif method == "median":
+            df[k].fillna(df[k].median(), inplace = True)
+    return df
+
+def get_census_income_agg(df):
+    return df.groupby(["nm_micro_regiao" ,"nm_meso_regiao"])[["empsetorcensitariofaixarendapopulacao"]].mean()
+
+def generate_additional_label(df, col_list):
+    for col in col_list:
+        df[col+"_label"] = 0
+        df.loc[~ df[col].isnull(), col+"_label"] = 1
+        df[col+"_label"] = df[col+"_label"].astype("bool")
+    return df
+
+def columns_to_boolean(df, col_list):
+    for col in col_list:
+        df[col] = df[col].astype("bool")
+    return df
+
+def main_pipeline(df):
+    """Runs extensively all dataframe formatting pipeline
+    
+    Arguments:
+        df {DataFrame} -- Full RAW Market Dataframe
+
+    Returns:
+        DataFrame -- Full DataFrame with features treated, nulls formatted and so and on
+    """
+    
+    df = fill_with_zero(df, "(funcionarios|veiculo)", True)
+    df = fill_by_dict(df, mapping_dicts.FILL_BINARY_N_CONTINUOUS)
+    df = fill_faturamento_estimado(df)
+    df = treat_datetime(df, "dt_situacao")
+    df = generate_additional_label(df, mapping_dicts.COLUMNS_WITH_ADDITIONAL_LABEL_FOR_NULL)
+    df = map_dataframe(df, mapping_dicts.MAP_TO_NUMERICAL_ENCODING)
```

Comment:
You could have used `.pipe` https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.pipe.html